lib/pages/chat_list/chat_list_header.dart

```diff
@@ -19,139 +18,112 @@ class ChatListHeader extends StatelessWidget {
-                  icon: Icon(
-                    controller.anySelectedRoomNotMuted
-                        ? Icons.notifications_off_outlined
-                        : Icons.notifications_outlined,
-                  ),
-                  tooltip: L10n.of(context)!.toggleMuted,
-                  onPressed: controller.toggleMuted,
-                ),
-              if (selectMode == SelectMode.select)
-                IconButton(
-                  icon: const Icon(Icons.archive_outlined),
-                  tooltip: L10n.of(context)!.archive,
-                  onPressed: controller.archiveAction,
-                ),
-            ],
-          ),
+          child: child,
         )
       ],
     );
   }
+
+  Widget selectModeWidgets(BuildContext context) {
+    return Row(
+      children: [
+        IconButton(
+          tooltip: L10n.of(context)!.cancel,
+          icon: const Icon(Icons.close_outlined),
+          onPressed: controller.cancelAction,
+          color: Theme.of(context).colorScheme.primary,
+        ),
+        Expanded(
+          child: Text(
+            controller.selectedRoomIds.length.toString(),
+            key: const ValueKey(SelectMode.select),
+          ),
+        ),
+        IconButton(
+          tooltip: L10n.of(context)!.toggleUnread,
+          icon: Icon(
+            controller.anySelectedRoomNotMarkedUnread
+                ? Icons.mark_chat_read_outlined
+                : Icons.mark_chat_unread_outlined,
+          ),
+          onPressed: controller.toggleUnread,
+        ),
+        IconButton(
+          tooltip: L10n.of(context)!.toggleFavorite,
+          icon: Icon(
+            controller.anySelectedRoomNotFavorite
+                ? Icons.push_pin_outlined
+                : Icons.push_pin,
+          ),
+          onPressed: controller.toggleFavouriteRoom,
+        ),
+        IconButton(
+          icon: Icon(
+            controller.anySelectedRoomNotMuted
+                ? Icons.notifications_off_outlined
+                : Icons.notifications_outlined,
+          ),
+          tooltip: L10n.of(context)!.toggleMuted,
+          onPressed: controller.toggleMuted,
+        ),
+        IconButton(
+          icon: const Icon(Icons.archive_outlined),
+          tooltip: L10n.of(context)!.archive,
+          onPressed: controller.archiveAction,
+        ),
+      ],
+    );
+  }
+
+  Widget normalModeWidgets(BuildContext context) => Row(
+        children: [
+          Expanded(
+            child: InkWell(
+              borderRadius: BorderRadius.circular(24.0),
+              onTap: onOpenSearchPage,
+              child: TextField(
+                controller: controller.searchChatController,
+                textInputAction: TextInputAction.search,
+                onChanged: controller.onSearchEnter,
+                enabled: false,
+                decoration: InputDecoration(
+                  filled: true,
+                  contentPadding: const EdgeInsets.all(0),
```

Comment:
does it have the same effect ?
`isCollapsed: true`

Reply:
i think its different.